nixos/modules/config/xdg/portal.nix

```diff
@@ -28,10 +40,17 @@ with lib;
 
     in mkIf cfg.enable {
 
+      assertions = [
+        { assertion = (cfg.gtkUsePortal -> cfg.extraPortals != []);
+          message = "Setting xdg.portal.gtkUsePortal to true without a XDG portal implementation in xdg.portal.extraPortals will cause things to break.";
```

Comment:
Could we nudge the user towards an option or a solution, instead of only admonishing them for their choice?